functions.cpp

```diff
@@ -408,15 +408,19 @@ namespace Sass {
       HSL hsl_color = rgb_to_hsl(rgb_color->r(),
                                  rgb_color->g(),
                                  rgb_color->b());
-      //Check saturation is not negative before saturate it
-      double hslcolorS = hsl_color.s;
+
+      double hslcolorS = hsl_color.s + amount->value();
```

Comment:
Can input be negative? I assume `rgb_to_hsl` will never return a colour with negative saturation, and if you pass a negative number to the `saturate()` function, an error will be thrown.